### ISSUE TITLE
PB: Book-State not saved on powerdown

### ISSUE THREAD

**Issue by mk-f** [koreader/koreader#3477]
* KOReader version: debug-v2015.11-1404-g15157536
* Device: PB Inkpad 2

#### Issue
It seems like putting the PB in sleep-mode does not update the Book-State: The PocketBook will sooner or later power down fully when in sleep-mode.
After booting again the page-count is lost, and the book starts at page one (or another, previously saved state). I made it a habit to exit the application before putting the PB to sleep, in this way the state is preserved.

#### Steps to reproduce
Open a book, skip some pages, put the PB to sleep, wait till it powers down (is it even supposed to do this? My PB-setting is 'Power down after 10 Minutes', but this is surely not whats happening - the power down comes much later - maybe it is in fact a crash?)

##### `crash.log` (if applicable)
[crash.log](https://github.com/koreader/koreader/files/1463823/crash.log)

**Comment by Frenzie** [comment]
We want to move the book state to being saved more often but first bookmarks and such will have to be migrated out. But I don't know what to make of this powering down business.

**Comment by chrox** [comment]
On pocketbook devices the [`EVT_BACKGROUND` is emitted](https://github.com/koreader/koreader/blob/71553a5190197d80c069895daa34c238bcf9c4f3/frontend/device/pocketbook/device.lua#L60) when the devices are going to sleep but [`Device:onPowerEvent`](https://github.com/koreader/koreader/blob/71553a5190197d80c069895daa34c238bcf9c4f3/frontend/device/generic/device.lua#L93) does not handle reader state saving. May be we can refactoring the flush setting logic from [uimanager](https://github.com/koreader/koreader/blob/71553a5190197d80c069895daa34c238bcf9c4f3/frontend/ui/uimanager.lua#L781-L785) to the generic device interface.

**Comment by farvardin** [comment]
I own a PocketBook 626 and I have the same problem since a few KoReader release. I think the first one I've tried 2 years ago was working correctly. Then, since a few releases, it looks like it's working at random (when I exited the app with the home button it saved my position at least, but if I forgot to exit sometimes it saved, other times it didn't), and for the latest one, it never works, even when I exit the app. 

I also noticed I could set a bookmark, and it appeared in the bookmark list, but when the PocketBook went into sleep mode or shut down, when I open again the book, the bookmarks all disappear.
